contract/services.py

```diff
@@ -75,15 +87,71 @@ def __evaluate_contract_valuation(self, contract_details_result):
         )
         return result_contract_valuation["data"]["total_amount"]
 
+    # TODO update contract scenario according to wiki page
+    @check_authentication
+    def update(self, contract):
+        try:
+            # check rights for contract / amendments
+            if not (self.user.has_perms(ContractConfig.gql_mutation_update_contract_perms) or self.user.has_perms(
+                    ContractConfig.gql_mutation_approve_ask_for_change_contract_perms)):
+                raise PermissionError("Unauthorized")
+            updated_contract = ContractModel.objects.filter(id=contract['id']).first()
+            # updatable scenario
```

Comment:
check status: if in “draft”/ “request for information“ / “counter“ only the  code cannot be changed,